framework/contracts/native/version-control/src/commands.rs

```diff
@@ -31,32 +35,95 @@ use crate::{
 };
 
 /// Add new Account to version control contract
-/// Only Factory can add Account
+/// Only Account can add itself.
 pub fn add_account(
     deps: DepsMut,
     msg_info: MessageInfo,
-    account_id: AccountId,
-    account_base: Account,
     namespace: Option<String>,
+    creator: String,
```

Comment:
Creator is never used, but to check that it's the ibc-host in case of remote accounts. So this seems to be a useless parameter

Reply:
How do we check that the ibc-host created the account if not for this parameter?

Reply:
Sorry my bad, this parameter can be trusted from the caller, so it's necessary, removing this comment